repository/src/main/java/com/epam/esm/dao/impl/GiftCertificateDaoImpl.java

```diff
@@ -0,0 +1,126 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.GiftCertificateDao;
+import com.epam.esm.dao.constant.EntityFieldsName;
+import com.epam.esm.dao.creator.QueryCreator;
+import com.epam.esm.dao.creator.criteria.Criteria;
+import com.epam.esm.dto.GiftCertificate;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaDelete;
+import javax.persistence.criteria.CriteriaQuery;
+
+import javax.persistence.criteria.Root;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * The type Gift certificate dao.
+ */
+@Repository
+public class GiftCertificateDaoImpl implements GiftCertificateDao<GiftCertificate> {
+    private final QueryCreator<GiftCertificate> criteriaCreator;
+    private final EntityManagerFactory factory;
+
+    /**
+     * Instantiates a new Gift certificate dao.
+     *
+     * @param criteriaCreator the criteria creator
+     * @param factory         the factory
+     */
+    @Autowired
+    public GiftCertificateDaoImpl(QueryCreator<GiftCertificate> criteriaCreator, EntityManagerFactory factory) {
+        this.criteriaCreator = criteriaCreator;
+        this.factory = factory;
+    }
+
+    @Override
+    public long insert(GiftCertificate giftCertificate) {
+        EntityManager em = factory.createEntityManager();
```

Comment:
Do we need to get the entity manager each time?

Reply:
No, we don't need to get the entity manager each time.